src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -2113,30 +2113,28 @@ private static void RemoveMeshGeometryByRange(List<(int start, int end)> vertexR
         /// <param name="l">line object</param>
         private static void RemoveLineGeometryByRange(List<(int start, int end)> verticesRange, LineGeometry3D l)
         {
-            //First sort the range data
+            //First sort the range data so that we remove the range from the end to the start.
             verticesRange.Sort();
             verticesRange.Reverse();
 
-            //track removed vertices to renumber indices index
-            var totalRemoved = 0;
-            
             //Remove already generated line geometry from render package
+
             foreach (var range in verticesRange)
             {
                 var i = range.start;
-                var c = range.end - range.start + 1;
-                l.Positions.RemoveRange(i, c);
-                l.Colors.RemoveRange(i, c);
-                totalRemoved += c;
+                var count = range.end - range.start + 1;
+                l.Positions.RemoveRange(i, count);
+                l.Colors.RemoveRange(i, count);
 
                 var firstIndicesIndex = l.Indices.IndexOf(range.start);
                 var indicesCount = l.Indices.IndexOf(range.end) - firstIndicesIndex + 1;
                 l.Indices.RemoveRange(firstIndicesIndex, indicesCount);
-            }
 
-            for (int i = 0; i < l.Indices.Count; i++)
-            {
-                l.Indices[i] -= totalRemoved;
+                //Reset the Indices values for the indices that were after the removed region
```

Comment:
Is this a correct statement : the value of l.Indices[ j ] is always smaller then the value of l.Indices[ j+1 ] ?
I'm thinking if there is a way to check the validity of Geometry3D at the end of this operation and throw is something is not right.

Reply:
Looked into this and will have to come back to it.  There isn't a very predictable try / catch situation upward.  I think we will have to come back with that.

Reply:
I added the assert in the tests but that is really what we want to eventually check at runtime.  indices should not be less then 0 or greater then vertex count.